/*
 * This file was automatically generated by EvoSuite
 * Wed Jul 06 18:45:40 GMT 2016
 */

package se.cag.labs.usermanager;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import se.cag.labs.usermanager.NewUser;
import se.cag.labs.usermanager.Token;
import se.cag.labs.usermanager.User;
import se.cag.labs.usermanager.UserManagerController;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class UserManagerController_ESTest extends UserManagerController_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      UserManagerController userManagerController0 = new UserManagerController();
      NewUser newUser0 = new NewUser();
      // Undeclared exception!
      try { 
        userManagerController0.registerNewUser(newUser0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("se.cag.labs.usermanager.UserManagerController", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      UserManagerController userManagerController0 = new UserManagerController();
      // Undeclared exception!
      try { 
        userManagerController0.logout((Token) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("se.cag.labs.usermanager.UserManagerController", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      UserManagerController userManagerController0 = new UserManagerController();
      User user0 = new User();
      // Undeclared exception!
      try { 
        userManagerController0.login(user0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("se.cag.labs.usermanager.UserManagerController", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      UserManagerController userManagerController0 = new UserManagerController();
      Token token0 = new Token();
      // Undeclared exception!
      try { 
        userManagerController0.getUserForToken(token0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("se.cag.labs.usermanager.UserManagerController", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      UserManagerController userManagerController0 = new UserManagerController();
      // Undeclared exception!
      try { 
        userManagerController0.getUsers();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("se.cag.labs.usermanager.UserManagerController", e);
      }
  }
}
